src/vs/platform/extensionManagement/node/extensionGalleryService.ts

```diff
@@ -724,7 +734,8 @@ export class ExtensionGalleryService implements IExtensionGalleryService {
 	}
 
 	loadCompatibleVersion(extension: IGalleryExtension, fromVersion: string = extension.version): Promise<IGalleryExtension> {
-		if (extension.version === fromVersion && extension.properties.engine && isEngineValid(extension.properties.engine)) {
```

Comment:
Whats the purpose of disabling the additional engine checks?

Reply:
added comments explaining why.